docs2/site/docs/getting-started/errors.md

```diff
@@ -200,3 +200,36 @@ For example, to show the stack traces for unhandled errors during development, y
 You can also write your own implementation of `IErrorInfoProvider`. For instance, you might want to override
 the numerical codes provided by GraphQL.NET for validation errors, or reveal stack traces
 only to logged-in administrators.
+
+## <a name="ValidationErrors"></a>Validation error reference list
+
+Here is a full list of validation errors produced by GraphQL.NET:
+
+Rule                         | Code                             | Number
+-----------------------------|----------------------------------|-----------
+UniqueOperationNames         | UNIQUE_OPERATION_NAMES           | 5.2.1.1
+LoneAnonymousOperation       | LONE_ANONYMOUS_OPERATION         | 5.2.2.1
+SingleRootFieldSubscriptions | SINGLE_ROOT_FIELD_SUBSCRIPTIONS  | 5.2.3.1
+FieldsOnCorrectType          | FIELDS_ON_CORRECT_TYPE           | 5.3.1
+OverlappingFieldsCanBeMerged | OVERLAPPING_FIELDS_CAN_BE_MERGED | 5.3.2
+ScalarLeafs                  | SCALAR_LEAFS                     | 5.3.3
+KnownArgumentNames           | KNOWN_ARGUMENT_NAMES             | 5.4.1
+UniqueArgumentNames          | UNIQUE_ARGUMENT_NAMES            | 5.4.2
+ProvidedNonNullArguments     | PROVIDED_NON_NULL_ARGUMENTS      | 5.4.2.1
+UniqueFragmentNames          | UNIQUE_FRAGMENT_NAMES            | 5.5.1.1
+KnownTypeNames               | KNOWN_TYPE_NAMES                 | 5.5.1.2
+FragmentsOnCompositeTypes    | FRAGMENTS_ON_COMPOSITE_TYPES     | 5.5.1.3
+KnownFragmentNames           | KNOWN_FRAGMENT_NAMES             | 5.5.1.4
+NoUnusedFragments            | NO_UNUSED_FRAGMENTS              | 5.5.1.4
+NoFragmentCycles             | NO_FRAGMENT_CYCLES               | 5.5.2.2
+PossibleFragmentSpreads      | POSSIBLE_FRAGMENT_SPREADS        | 5.5.2.3
+ArgumentsOfCorrectType       | ARGUMENTS_OF_CORRECT_TYPE        | 5.6.1
```

Comment:
ARGUMENTS_OF_CORRECT_TYPE and DEFAULT_VALUES_OF_CORRECT_TYPE both have 5.6.1

Reply:
Let me know what numbers to use; I'll make the appropriate changes.

Reply:
Looks like 5.6.1 really covers 2 cases for value literals - argument literal and default values for variables.

Reply:
By the way, could you check `DefaultValuesOfCorrectType` rule: 
```c#
 if (inputType is NonNullGraphType nonNullType && defaultValue != null)  <--- I don't undestand this condition
                    {
                        context.ReportError(new ValidationError(
                            context.OriginalQuery,
                            "5.7.2",
                            BadValueForNonNullArgMessage(
                                name,
                                context.Print(nonNullType),
                                context.Print(nonNullType.ResolvedType)),
                            defaultValue));
                    }
```

Reply:
The code you pointed out covers the test `no_required_variables_with_default_values`:

```gql
query UnreachableDefaultValues($a: Int! = 3, $b: String! = ""default"") {
  dog { name }
}
```

Where the failure is that a non-null type (`Int!`) requires the argument to be provided and so should not specify a default value.

Reply:
I think it's wrong. I've already seen a similar situation for argument default values. And I told you about it. I think the situation is similar with variables, although I cannot easily find words about this in the specification. I talk about that:
> Arguments can be required. An argument is required if the argument type is non‐null and does not have a default value. Otherwise, the argument is optional.

Let's make a separate issue for this, so as not to keep this PR opened.

Reply:
I agree it's probably wrong.  I'm happy to set it however you suggest.  It seems to me like this is more of a warning than an error that violates spec.  Someone coded it, so perhaps it was explicitly an error in a prior version of the spec; I don't know.  Anyway, I'll open an issue for it.

Reply:
I don't know too.